src/main/java/org/springframework/data/elasticsearch/repository/support/MappingElasticsearchEntityInformation.java

```diff
@@ -47,9 +47,6 @@ public MappingElasticsearchEntityInformation(ElasticsearchPersistentEntity<T> en
 	public MappingElasticsearchEntityInformation(ElasticsearchPersistentEntity<T> entity, String indexName, String type) {
 		super(entity);
 
```

Comment:
the `indexName` and  `type` may be null and later be resolved using the `entityMetadata` but then a check should be added here that `entity` is not null.

Reply:
The non-null check is already done in the superclass, so IMO it is not necessary to do it twice.